File: app/src/main/java/io/github/howiezuo/unsplash/feature/detail/DetailComponent.java
package io.github.howiezuo.unsplash.feature.detail;

import dagger.Component;
import io.github.howiezuo.unsplash.app.ActivityScope;

@ActivityScope
@Component(
        modules = {
                DetailViewModelModule.class
        }
)
public interface DetailComponent {

    void inject(DetailActivity activity);

}

File: app/src/main/java/io/github/howiezuo/unsplash/feature/detail/DetailComponent.kt
package io.github.howiezuo.unsplash.feature.detail

import dagger.Component
import io.github.howiezuo.unsplash.app.ActivityScope


@ActivityScope
@Component(
        modules = arrayOf(
                DetailViewModelModule::class
        )
)
interface DetailComponent {

    fun inject(activity: DetailActivity)

}
